File: An 2/Sisteme de Gestiunea Bazelor de Date/problema11ClientServer/Networking/utils/AbstractServer.cs
using System;
using System.Net;
using System.Net.Sockets;

namespace Networking.utils
{
    public abstract class AbstractServer
    {
        private TcpListener server;
        private string host;
        private int port;
        
        public AbstractServer(string host, int port)
        {
            this.host = host;
            this.port = port;
        }
        
        public void Start()
        {
            var adr = IPAddress.Parse(host);
            var ep = new IPEndPoint(adr, port);
            server = new TcpListener(ep);
            server.Start();
            while (true)
            {
                Console.WriteLine("Waiting for clients ...");
                var client = server.AcceptTcpClient();
                Console.WriteLine("Client connected ...");
                processRequest(client);
            }
        }
        
        public abstract void processRequest(TcpClient client);
    }
}

File: An 2/Medii de Proiectare si Programare/SERVER_CLIENT_MPP/Networking/src/main/java/com/example/Utils/AbstractServer.java
package com.example.Utils;

import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;


public abstract class AbstractServer {
    private final int port;
    private ServerSocket server = null;
    public AbstractServer(int port){
        this.port = port;
    }

    public void start() throws ServerException {
        try{
            server = new ServerSocket(port);
            while (true){
                System.out.println("Waiting for clients ...");
                Socket client = server.accept();
                System.out.println("Client connected ...");
                processRequest(client);
                System.out.println("IA LA MUE \n \n \n \n \n \n \n \n \n \n \n \n \n \n \n \n \n \n \n");
            }
        } catch (IOException e) {
            throw new ServerException("Starting server errror ",e);
        } finally {
            stop();
        }
    }

    protected abstract void processRequest(Socket client);

    public void stop() throws ServerException {
        try {
            server.close();
        } catch (IOException e) {
            throw new ServerException("Closing server error ", e);
        }
    }
}